### PR TITLE
Improve Error message for Angular.json file not found

### LINKED ISSUE #22351: storybook init: No `angular.json` file found
Error location: [angular generator → AngularJSON](https://github.com/storybookjs/storybook/blob/next/code/lib/cli/src/generators/ANGULAR/helpers.ts#L36)

Suggested work:

This is related to projects that try to initialize Storybook but do not have an `angular.json` file *in the current working directory.* We should investigate what exactly is happening, whether this is a correct scenario or if we could improve it. For instance, we could use [find-up](https://www.npmjs.com/package/find-up) to detect the angular.json file rather than `readSync` which only looks for files at the current working directory.

If there is no need to change the behavior because the error is legit, then we should at least improve the error message to provide a link to an FAQ entry (that should be created as part of this task) in the Storybook documentation website.

### PR DESCRIPTION
**Pull request by valentinpalkovic**
Closes https://github.com/storybookjs/storybook/issues/22351

## What I did

Improved error message and updated faq

## Checklist

- [ ] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [x] Make sure to add/update documentation regarding your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [x] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -55,7 +60,7 @@ export default {
 };
 ```
 
-### How can I opt-out of Angular ngcc?
+## How can I opt-out of Angular ngcc?
```
Comment (yannbf) [inline docs/faq.md:63]: was the hierarchy level change intentional? or auto-save kind of formatting

Comment (valentinpalkovic) [reply docs/faq.md:63]: Yes it was. I streamlined it. It was chaotic.

File: docs/faq.md [hunk 2]
```diff
@@ -34,7 +35,11 @@ Here are some answers to frequently asked questions. If you have a question, you
 - [Why isn't Storybook's test runner working?](#why-isnt-storybooks-test-runner-working)
 - [How does Storybook handle environment variables?](#how-does-storybook-handle-environment-variables)
 
-### How can I opt-out of Angular Ivy?
+## Error: No angular.json file found
+
+Storybook supports Angular multi-project workspace. You can setup Storybook for each project in the workspace. When running npx storybook@latest init you will be asked for which project Storybook should be set up. Essentially, during initialization, the .storybook folder will be created and the angular.json will be edited to add the Storybook configuration for the selected project. Therefore it is important to always run the init command from the root of the workspace, where the angular.json file is located.
```
Comment (yannbf) [inline docs/faq.md:40]: What about:

```suggestion
Storybook can be set up for both single-project and multi-project Angular workspaces. To set up Storybook for a project, run `npx storybook@latest init` at the root of the workspace where the `angular.json` file is located. During initialization, the `.storybook` folder will be created and the `angular.json` file will be edited to add the Storybook configuration for the selected project. It's important to run the command at the root level to ensure that Storybook detects all projects correctly.
```

File: code/lib/cli/src/generators/ANGULAR/helpers.ts [hunk 3]
```diff
@@ -29,11 +27,9 @@ export class AngularJSON {
 
   constructor() {
     if (!fs.existsSync(ANGULAR_JSON_PATH)) {
-      commandLog(
-        'An angular.json file was not found in the current directory. Storybook needs it to work properly.'
+      throw new Error(
+        'An angular.json file was not found in the current working directory. Storybook needs it to work properly. Please go to https://storybook.js.org/docs/angular/faq#error-no-angularjson-file-found for troubleshooting'
```
Comment (yannbf) [inline code/lib/cli/src/generators/ANGULAR/helpers.ts:31]: What about:
```suggestion
        'An angular.json file was not found in the current working directory. Storybook needs it to work properly, so please rerun the command at the root of your project, where the angular.json file is located. More info: https://storybook.js.org/docs/angular/faq#error-no-angularjson-file-found'
```

**Review by yannbf — APPROVED** [review]
Awesome <3

**Pull request merged**